File: backend/emne.js
import Notat from "./notat.js";

export class Emne {
  constructor(navn,emnekode) {
    this._navn = navn; //str
    this._emnekode = emnekode; //str
    this._notater = []; //array (liste)
  }

  hentNavn(){
    return this._navn;
  }

  hentEmnekode(){
    return this._emnekode;
  }
  hentNotater(){
    return this._notater;
  }

  leggTilNotat(notat){
    this._notater.push(notat);
  }

  toString(){
    return this._emnekode + ": " + this._navn;
  }

}

File: backend/emne.py
from notat import Notat

class Emne:
    def __init__(self, navn, emnekode):
        self._navn = navn #str
        self._emnekode = emnekode #str
        self._notater = [] #list , kan bruke dict om jeg vil kunne hente spesifikke notater

    def hentNavn(self):
        return self._navn

    def hentEmnekode(self):
        return self._emnekode

    def hentNotater(self):
        return self._notater

    def hentNotat(self, link):
        for notat in self._notater:
            if notat.hentLink() == link:
                return notat

    def leggTilNotat(self, notat):
        self._notater.append(notat)

    def sjekkOmNotatFinnes(self, linkNotat):
        for notat in self._notater:
            if linkNotat == notat.hentLink():
                return True
        return False

    def lastUt(self):
        return self._navn + ";" + self._emnekode + ";"

    def __str__(self):
        return self._emnekode + ": " + self._navn
